sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/CosmosEncryptionAsyncClient.java

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.caches.AsyncCache;
+import com.azure.cosmos.models.ClientEncryptionPolicy;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyProperties;
+import com.microsoft.data.encryption.cryptography.EncryptionKeyStoreProvider;
+import reactor.core.publisher.Mono;
+
+/**
+ * CosmosClient with Encryption support.
+ */
+public class CosmosEncryptionAsyncClient {
```

Comment:
Can you please provide sample of the user code in the PR description?

Reply:
Done